kt-code-generators/src/main/kotlin/io/vrap/codegen/kt/languages/java/model/JavaObjectTypeRenderer.kt

```diff
@@ -0,0 +1,197 @@
+        } else {
+            "private ${this.type.toVrapType().simpleName()} ${if (this.isPatternProperty()) "values" else this.name};"
+        }
+
+    }
+
+    fun ObjectType.toBeanFields() = this.properties
+            .filter { it.name != this.discriminator }
+            .map { it.toJavaField() }.joinToString(separator = "\n\n")
+
+    fun ObjectType.setters() = this.properties
+            //Filter the discriminators because they don't make much sense the generated bean
+            .filter { it.name != this.discriminator }
+            .map { it.setter() }
+            .joinToString(separator = "\n\n")
+
+
+    fun ObjectType.getters() = this.properties
+            //Filter the discriminators because they don't make much sense the generated bean
+            .filter { it.name != this.discriminator }
+            .map { it.getter() }
+            .joinToString(separator = "\n\n")
+
+    fun Property.isPatternProperty() = this.name.startsWith("/") && this.name.endsWith("/")
+
+    fun Property.setter(): String {
+        return if (this.isPatternProperty()) {
+
+            """
+            |@JsonAnySetter
+            |public void setValue(String key, ${this.type.toVrapType().simpleName()} value) {
+            |    if (values == null) {
+            |        values = new HashMap<>();
+            |    }
+            |    values.put(key, value);
+            |}
+            """.trimMargin()
+        } else {
+            """
+            |public void set${this.name.capitalize()}(final ${this.type.toVrapType().simpleName()} ${this.name}){
+            |   this.${this.name} = ${this.name};
+            |}
+            """.trimMargin()
+        }
+    }
+
+    fun Property.getter(): String {
+        return if (this.isPatternProperty()) {
+
+            """
+            |${this.validationAnnotations()}
+            |${this.type.toJavaComment()}
+            |@JsonAnyGetter
+            |public Map<String, ${this.type.toVrapType().simpleName()}> values() {
+            |    return values;
+            |}
+            """.trimMargin()
+        } else {
+            """
+            |${this.type.toJavaComment()}
+            |${this.validationAnnotations()}
+            |@JsonProperty("${this.name}")
+            |public ${this.type.toVrapType().simpleName()} get${this.name.capitalize()}(){
+            |   return this.${this.name};
+            |}
+        """.trimMargin()
+        }
+    }
+
+    fun Property.validationAnnotations(): String {
+        val validationAnnotations = ArrayList<String>()
+        if (this.required != null && this.required!!) {
+            validationAnnotations.add("@NotNull")
+        }
+        if (CascadeValidationSwitch.doSwitch(this.type)) {
+            validationAnnotations.add("@Valid")
+        }
+        return validationAnnotations.joinToString(separator = "\n")
+    }
+
+
+
+
+    private object CascadeValidationSwitch : TypesSwitch<Boolean>() {
```

Comment:
Is this switch responsible to add the `@Valid` annotation? How about renaming it to `CascadeValidationCheck`?

Reply:
sounds good 👍